contrib/checkbox-ce-oem/checkbox-provider-ce-oem/data/video-codec-test-confs/genio-1200.json

```diff
@@ -1,90 +1,133 @@
+      "color_spaces": ["NV12", "NV21", "ARGB", "RGB", "I420", "YV12"],
+      "source_format": "mp4"
+    },
+    {
+      "decoder_plugin": "v4l2h265dec",
+      "resolutions": [{ "width": 320, "height": 320 }],
+      "color_spaces": ["NV12", "NV21", "ARGB", "RGB", "I420", "YV12"],
+      "source_format": "mp4"
+    },
+    {
+      "decoder_plugin": "v4l2vp8dec",
+      "resolutions": [{ "width": 320, "height": 320 }],
+      "color_spaces": ["YUV"],
+      "source_format": "webm"
     },
-    "gst_v4l2_video_decoder_performance_fakesink": [
-        {
-            "decoder_plugin": "v4l2h264dec",
-            "golden_sample_file": "H264_8Bit_High@L6.1_3840X2160_90Fps_160Mbps_AACLC-10sec.mp4",
-            "minimum_fps": 90,
-            "enable_performance_mode": true
-        },
-        {
-            "decoder_plugin": "v4l2h265dec",
-            "golden_sample_file": "HEVC_8Bit_High@L5.2_3840X2160_90Fps_160Mbps_AACLC-10sec.mp4",
-            "minimum_fps": 90,
-            "enable_performance_mode": true
-        },
-        {
-            "decoder_plugin": "v4l2mpeg4dec",
-            "golden_sample_file": "MPEG-4_1920x1080_60fps_60Mbps.mp4",
-            "minimum_fps": 60,
-            "enable_performance_mode": true
-        },
-        {
-            "decoder_plugin": "v4l2vp8dec",
-            "golden_sample_file": "VP8_1920x1080_60fps_40Mbps.webm",
-            "minimum_fps": 60,
-            "enable_performance_mode": true
-        },
-        {
-            "decoder_plugin": "v4l2vp9dec",
-            "golden_sample_file": "VP9_3840x2160_90fps_120Mbps.webm",
-            "minimum_fps": 90,
-            "enable_performance_mode": true
-        }
+    {
+      "decoder_plugin": "v4l2vp9dec",
+      "resolutions": [{ "width": 320, "height": 320 }],
+      "color_spaces": ["YUV"],
+      "source_format": "webm"
+    }
+  ],
+  "gst_v4l2_audio_video_synchronization": {
+    "video_sinks": {
+      "on_desktop": "waylandsink",
+      "on_server": "kmssink connector-id=32 driver-name=mediatek",
+      "on_core": "kmssink connector-id=32 driver-name=mediatek"
+    },
+    "cases": [
+      {
+        "decoder_plugin": "v4l2h264dec",
+        "golden_sample_files": [
+          {
+            "file_name": "480p.mp4",
+            "capssetter_pipeline": ""
+          },
+          {
+            "file_name": "720p.mp4",
+            "capssetter_pipeline": ""
+          },
+          {
+            "file_name": "1080p.mp4",
+            "capssetter_pipeline": ""
+          },
+          {
+            "file_name": "4K.mp4",
+            "capssetter_pipeline": "capssetter replace=true caps='video/x-h264, stream-format=(string)byte-stream, alignment=(string)au, level=(string)5.2, profile=(string)main, width=(int)3840, height=(int)2160, framerate=(fraction)24/1, pixel-aspect-ratio=(fraction)1/1, colorimetry=(string)bt2020, chroma-format=(string)4:2:0, bit-depth-luma=(uint)8, bit-depth-chroma=(uint)8, parsed=(boolean)true'"
```

Comment:
It looks like all the `capssetter_pipeline`s are using `replace=true` so I think we could store only the caps and interpolate it on the fly. For example we can just store
```json
"caps": "video/x-h264, stream-format=(string)byte-stream, alignment=(string)au, level=(string)5.2, profile=(string)main, width=(int)3840, height=(int)2160, framerate=(fraction)24/1, pixel-aspect-ratio=(fraction)1/1, colorimetry=(string)bt2020, chroma-format=(string)4:2:0, bit-depth-luma=(uint)8, bit-depth-chroma=(uint)8, parsed=(boolean)true'"
```
and use it like this:
```py
json_file = json.load(...) # load this json somewhere 
execute_command("gst-launch-1.0 ! v4l2src ! capssetter replace=true caps='{}'"\
    .format(json_file["cases"][0]["golden_sample_files"][3]["caps"])
)
```
Totally optional, could be helpful if you need the caps string itself instead of the entire element

Reply:
We won't access the JSON file in each gst implement script like gst_encode_psnr.py directly. Only load it in gst_resources_generator.py.

Reply:
Another thing is we will build the gstreamer pipeline in seach gst implementation in the future. Therefore, we won't record any pipeline in JSON file.